### PR TITLE
[CLE GA A2.1] Moving to opening Case Search Index for all domains

### PR DESCRIPTION
**Pull request by biyeun**
## Technical Summary
As part of the GA of the Case List Explorer...

This PR introduces a new model called `DomainsNotInCaseSearchIndex`, which is a temporary model to keep track of including legacy domain case data in the Case Search Index.

In the migration introducing this model, we iterate through all existing domains that are not yet included in the Case Search Index, count the number of cases in that domain, and add a `DomainsNotInCaseSearchIndex` record to note that this domain is not yet part of this index.

Changes are then made to the case search pillow to ensure that ALL domains EXCEPT the ones with a `DomainsNotInCaseSearchIndex` record are now being included in the index.

There will be another PR to systematically migrate the remaining domains over to the Case Search Index based on the number of cases (`estimated_size` in `DomainsNotInCaseSearchIndex`). Prior to this, we will run some tests to evaluate the impact of larger migrations on our system and establish a way to do the migration (perhaps a periodic task scheduled during a low usage window) to migrate the remaining domains.

This PR completes the steps on the following tickets:
https://dimagi-dev.atlassian.net/browse/SAAS-14445
https://dimagi-dev.atlassian.net/browse/SAAS-14446

### Safety story
Tests should cover any issues.

### Automated test coverage
Yes

### QA Plan
Not needed.

### Migrations

- [ ] The migrations in this code can be safely applied first independently of the code
^ note that a this PR can't be rolled back in a direct way, but the migration is reversible in the event we need to.

### Rollback instructions

- [ ] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/pillows/case_search.py [hunk 1]
```diff
@@ -205,7 +203,7 @@ def build(self):
                 domains = [domain]
             else:
                 # return changes for all enabled domains
-                domains = domains_needing_search_index()
+                domains = _domains_needing_search_index()
```
Comment (AmitPhulera) [inline corehq/pillows/case_search.py:206]: Ongoing work removes the use of transform functions and has put them into the adapter itself. 
So we need to update `_from_dict` function of `corehq.apps.es.case_search.ElasticCaseSearch`

**Review by calellowitz — APPROVED** [review]
Besides Amit's comment looks good

File: corehq/pillows/case_search.py (hunk 1 above)
Comment (biyeun) [reply corehq/pillows/case_search.py:202]: just an FYI that Amit and I chatted offline and realized that he had misread the diff and that the transform function was not modified in this process  :)

Comment (AmitPhulera) [reply corehq/pillows/case_search.py:202]: Yep. No blockers from my side.

**Review by AmitPhulera — APPROVED** [review]

**Pull request merged**